shared/src/simulation/behaviors/automatically-distribute-vehicles.ts

```diff
@@ -0,0 +1,243 @@
+                        );
+
+                        // distribute
+
+                        const regionsOrderedByNeed = Object.keys(
+                            cloneDeepMutable(
+                                behaviorState.distributionDestinations
+                            )
+                        );
+                        regionsOrderedByNeed.sort(
+                            (a, b) =>
+                                Object.values(
+                                    behaviorState.remainingInNeed
+                                ).reduce((c, d) => c + (d[b] ? 1 : 0), 0) -
+                                Object.values(
+                                    behaviorState.remainingInNeed
+                                ).reduce((c, d) => c + (d[a] ? 1 : 0), 0)
+                        );
+
+                        const vehiclesToBeSent: {
+                            [region: UUID]: { [vehicletype: string]: 1 };
+                        } = {};
+
+                        Object.entries(behaviorState.remainingInNeed).forEach(
+                            ([vehicleType, regionsInNeed]) => {
+                                if (
+                                    (behaviorState.distributedRounds[
+                                        vehicleType
+                                    ] ?? 0) >=
+                                    (behaviorState.distributionLimits[
+                                        vehicleType
+                                    ] ?? 0)
+                                ) {
+                                    return;
+                                }
+
+                                Object.keys(regionsInNeed).forEach((region) => {
+                                    if (!vehiclesToBeSent[region]) {
+                                        vehiclesToBeSent[region] = {};
+                                    }
+                                    vehiclesToBeSent[region]![vehicleType] = 1;
+                                });
+                            }
+                        );
+
+                        regionsOrderedByNeed.forEach((region) => {
+                            if (!vehiclesToBeSent[region]) {
+                                return;
+                            }
+                            addActivity(
+                                simulatedRegion,
+                                TransferVehiclesActivityState.create(
+                                    nextUUID(draftState),
+                                    region,
+                                    'automatic-distribution',
+                                    VehicleResource.create(
+                                        cloneDeepMutable(
+                                            vehiclesToBeSent[region]!
+                                        )
+                                    )
+                                )
+                            );
+                        });
+                    }
```

Comment:
> suggest calculating, how many regions can be served and only creating the activities for them.

This is/should only be done by the Transfer Vehicles Activity

Reply:
The order activities are executed in, is not enforced. Your prioritization therefore does not really apply.

Reply:
Well w/o enforcement of activity execution order there is no real way to do it w/o double calculation what we wanted to avoid.  And I think while not explicitly enforced the activities are called in Order of addition to the state so it works

Reply:
the correct way would be to have another behavior that does the things the activity does so order could be preserved